### PR TITLE
enhance the ability of add

### PR DESCRIPTION
**Pull request by wuyudi**
### **Describe your change:**

enhance the ability of add

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
I am not a fan of the proposed change.  The current code allows the addition of multiple matrices while the proposed code only allows the addition of two matrices.  Why is the constraining of capabilities useful?

File: matrix/matrix_operation.py [hunk 1]
```diff
@@ -151,7 +155,8 @@ def _shape(matrix: List[list]) -> list:
     return list((len(matrix), len(matrix[0])))
 
 
-def _verify_matrix_sizes(matrix_a: List[list], matrix_b: List[list]) -> Tuple[list]:
+def _verify_matrix_sizes(
+        matrix_a: List[list], matrix_b: List[list]) -> Tuple[list]:
     shape = _shape(matrix_a)
```
Comment (cclauss) [inline matrix/matrix_operation.py:160]: Don't lines 159 and 160 need to have identical indentation?

Comment (wuyudi) [reply matrix/matrix_operation.py]: fixed.

Comment (wuyudi) [reply matrix/matrix_operation.py]: E125 continuation line with same indent as next logical line.

Comment (cclauss) [reply matrix/matrix_operation.py]: Line 159

File: matrix/matrix_operation.py [hunk 2]
```diff
@@ -5,16 +5,20 @@
...
     [[3, 5], [7, 9]]
     >>> add([[1.2,2.4],[3,4]],[[2,3],[4,5]])
     [[3.2, 5.4], [7, 9]]
+    >>> add([[1, 2], [4, 5]], [[3, 7], [3, 4]], [[3, 5], [5, 7]])
+    [[7, 14], [12, 16]]
     """
-    if _check_not_integer(matrix_a) and _check_not_integer(matrix_b):
-        _verify_matrix_sizes(matrix_a, matrix_b)
-        return [[i + j for i, j in zip(m, n)] for m, n in zip(matrix_a, matrix_b)]
+    if all(map(_check_not_integer, matrix_s)):
```
Comment (cclauss) [inline matrix/matrix_operation.py:17]:
```suggestion
    if all(_check_not_integer(m) for m in matrix_s):
```
As discussed in CONTRIBUTING.md.

**Comment by cclauss** [comment]
This looks good to me...  It would be helpful to get another reviewer.

**Pull request merged**